apps/api/src/app/rate-limiting/guards/throttler.guard.ts

```diff
@@ -165,18 +164,14 @@ export class ApiRateLimitInterceptor extends ThrottlerGuard implements NestInter
 
   private isAllowedAuthScheme(context: ExecutionContext): boolean {
     const req = context.switchToHttp().getRequest();
-    const authorization = req.headers[RateLimitHeaderKeysEnum.AUTHORIZATION.toLowerCase()];
-    if (!authorization) {
-      return false;
-    }
+    const authScheme = req.authScheme;
 
-    return ALLOWED_AUTH_SCHEMES.some((scheme) => req.authScheme === scheme);
+    return ALLOWED_AUTH_SCHEMES.some((scheme) => authScheme === scheme);
```

Comment:
What do you think about creating an auth scheme enum so it would be more transparent what scheme we have, this way by looking at the allowed scheme list we could figure out that Bearer is not allowed just by looking at the code.

```
export enum AuthSchemeEnum {
  API_KEY = 'ApiKey',
  BEARER = 'Bearer',
  NONE = 'None',
}

export const ALLOWED_AUTH_SCHEMES = [AuthSchemeEnum.API_KEY];
```

Reply:
I considered this, however we have the `UserAuthGuard` (formerly `JwtAuthGuard`) living in `application-generic` right now, with strategies living in `api`. These should all really be in the same place.

My suggestion would be to bring the `UserAuthGuard` back to `api`, and have both the guard and strategies co-located. We can move them all at once in the future if we need common API packages as part of API microservice direction or otherwise.

Wdyt @djabarovgeorge ?

Reply:
And yes, we 100% should create this enum!

Reply:
I haven't consolidated auth strategies and guards in this PR, however I have created an `auth.types` file in the `shared` lib to contain the auth strategies and auth scheme enums. Everywhere is now updated to use those 🎉

Reply:
> I considered this, however we have the `UserAuthGuard` (formerly `JwtAuthGuard`) living in `application-generic` right now, with strategies living in `api`. These should all really be in the same place.

Thanks for sharing your thoughts on the separation of the code, I agree that this guard's responsibility is to be used in the API domain but I believe that the JwtAuthGuard was moved to application-generic because we reuse it in the enterprise package controllers.